app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,19 @@
+class LikesController < ApplicationController
+  def create
+    @user = current_user
+    @post = Post.find(params[:post_id])
+    if liked?
+      redirect_to user_post_path(@user.id, @post.id)
+    else
+      @like = @user.likes.new
+      @like.author_id = @user.id
+      @like.post_id = @post.id
+      @like.save
+    end
+    redirect_to user_post_path(@user.id, @post.id)
+  end
+
+  def liked?
+    Like.where(author_id: @user.id, post_id: @post.id).exists?
+  end
```

Comment:
- [ ] KIndly solve this error that occurs when a post is liked 
![image](https://user-images.githubusercontent.com/65068771/187003113-de6d2d98-ef17-4177-b0b4-e163dcd36e5a.png)